File: EPI/12.5/typescript/12.5_test.ts
import sqrt from './12.5';

declare function require(name: string): any;
const assert = require('assert');

const tests: {n: number, sqrt: number}[] = [
    {n: 0, sqrt: 0},
    {n: 1, sqrt: 1},
    {n: 2, sqrt: 1},
    {n: 3, sqrt: 1},
    {n: 4, sqrt: 2},
    {n: 15, sqrt: 3},
    {n: 16, sqrt: 4},
    {n: 17, sqrt: 4},
    {n: 99, sqrt: 9},
];

for (let i = 0; i < 1000; i++) {
    const n = Math.floor(Math.random() * Number.MAX_SAFE_INTEGER);
    tests.push({n, sqrt: Math.floor(Math.sqrt(n))});
}

for (const test of tests) {
    const actual = sqrt(test.n);
    assert.strictEqual(actual, test.sqrt, `For number ${test.n}, ` +
        `expected integer square root to be ${test.sqrt}, but was ${actual}`);
}

File: EPI/Searching/Computer the integer square root/swift/12.5_test.swift
import Darwin

var tests: [(value: Int, root: Int)] = [
    (
        value: 0,
        root: 0
    ),
    (
        value: 1,
        root: 1
    ),
    (
        value: 2,
        root: 1
    ),
    (
        value: 3,
        root: 1
    ),
    (
        value: 4,
        root: 2
    ),
    (
        value: 8,
        root: 2
    ),
    (
        value: 9,
        root: 3
    ),
    (
        value: 120,
        root: 10
    ),
    (
        value: 121,
        root: 11
    ),
    (
        value: Int(Int32.max),
        root: 46340
    )
]

if Int64(Int.max) == Int64.max {
    tests.append((
        value: Int(Int64.max),
        root: 3037000499
    ))
}

for test in tests {
    let actual = root(test.value)
    guard actual == test.root else {
        print("For test value \(test.value), expected root to be \(test.root), but was \(actual)")
        exit(1)
    }
}
